fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -2689,4 +2693,28 @@ public void addFinishedQueryDetail() {
         QueryDetailQueue.addQueryDetail(queryDetail);
     }
 
+    public int getExecTimeout() {
+        if (parsedStmt instanceof DmlStmt) {
+            int timeout = ((DmlStmt) parsedStmt).getTimeout();
+            return timeout != -1 ? timeout : context.getSessionVariable().getInsertTimeoutS();
+        }
+        if (parsedStmt instanceof CreateTableAsSelectStmt) {
+            InsertStmt insertStmt = ((CreateTableAsSelectStmt) parsedStmt).getInsertStmt();
+            int timeout = insertStmt.getTimeout();
+            return timeout != -1 ? timeout : context.getSessionVariable().getInsertTimeoutS();
+        }
+        return context.getSessionVariable().getQueryTimeoutS();
+    }
+
+    public String getExecType() {
+        if (parsedStmt instanceof InsertStmt || parsedStmt instanceof CreateTableAsSelectStmt) {
```

Comment:
Why not implement it as a method in StatementBase?
And provide a StmtExecType class with DML_INSERT, DML_UPDATE, DML_DELETE, DCL_QUERY